### PR TITLE
Pagination theme prop

### PR DESCRIPTION
**Pull request by GCHQ-Developer-530**
## Summary of the changes
Replace appearance prop with theme and monochrome props on pagination to make use of light and dark
theme colours

BREAKING CHANGE: Appearance prop removed from pagination component and replaced with theme and
monochrome

## Related issue
Working towards #1214 

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 
- [x] Compare performance of modified components against develop using Performance addon in React Storybook.

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.
- [x] Props/slots can be updated after initial render.

### CODE REVIEW

**Comment by GCHQ-Developer-530** [comment]
> @GCHQ-Developer-530 I’m a bit curious, why some PRs are against v3 develop and not develop?
> 
> The cherry pick pr only works if you use develop as the commit branch.

Some PRs are only raised against v3 develop as they include breaking changes that we don't want to deploy to the customer yet. The cherry pick PR should take the work that's been done in develop (bug fixes, accessibility errors etc. that we want the customer to have now) and put that in v3 develop so that they're in sync.

**Comment by evenstensberg** [comment]
Okay, thanks for explaining. The strategy is then to force push the
v3/develop branch when due? I imagine there will be a lot of merge
conflicts, just so you know.


tir. 5. nov. 2024 kl. 15:28 skrev GCHQ-Developer-530 <
***@***.***>:

**Review by gd2910 — DISMISSED** [review]

**Review by ad9242 — CHANGES_REQUESTED** [review]
can we update the pagination bar React storybook? I think the Appearance story can be removed and the playground needs the them & monochrome props adding

File: packages/canary-web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx [hunk 1]
```diff
@@ -402,9 +408,24 @@ export class PaginationBar {
     );
   };
 
+  private handleButtonAppearance = () => {
+    if (this.monochrome) {
+      if (
+        this.theme === "dark" ||
+        (this.theme === "inherit" &&
+          window.matchMedia("(prefers-color-scheme: dark)").matches)
```
[the remark this answers is not in the document]
Comment (ad9242) [reply packages/canary-web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx:416]: Would be nice to make the dark mode check a helpers func as it already does similar in ic-theme. But may need to do in both web-components & canary-web-component helpers.ts

[the remark this answers is not in the document]
Comment (ad9242) [reply packages/canary-web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx:416]: actually not required as this code is temporary until ic-button supports theme prop

**Review by ad9242 — APPROVED** [review]

**Pull request merged**